### PR TITLE
Use primitive type instead of constructor type

### PR DESCRIPTION
**Pull request by onsummer**
Make types more precision.

This PR change constructor type to primitive type, e.g.

```diff
/**
 * Removes a key-value pair from the collection.
 *
- * @param {String|Number} key The key to be removed.
- * @returns {Boolean} True if it was removed, false if the key was not in the collection.
+ * @param {string|number} key The key to be removed.
+ * @returns {boolean} True if it was removed, false if the key was not in the collection.
 */
AssociativeArray.prototype.remove = function (key) {
```

TODO:
- [x] Use primitive type instead of Constructor
- [x] Update CHANGES.md

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @onsummer!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/ts-type-enhance/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by ggetz** [comment]
Hi @onsummer, thanks for the effort to fix up some typing issues!

I would appreciate if you could explain in a bit more detail here what changes you are making to the tooling, and why.

**Comment by onsummer** [comment]
@ggetz OK, poor for my English, I'll try to explain by edit above as concisely as possible

**Comment by onsummer** [comment]
@ggetz PTAL. I found that adding additional types (eg. callback functions' type) may require a new PR, what do you think?

**Comment by ggetz** [comment]
@onsummer, yes. I think it would be helpful for review to separate the tooling changes from the type updates. Thanks!

**Comment by ggetz** [comment]
Thanks @onsummer! 

One thing I'm still not clear on is why `jsdoc-tsimport-plugin` is needed. I don't see where we are using the `import` syntax. Could  you please explain?

**Comment by onsummer** [comment]
@ggetz Sorry, forget to remove that. This dependency should appearance in next PR that will improve types by using `import()` syntax.

File: packages/engine/Source/Renderer/ShaderBuilder.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ Object.defineProperties(ShaderBuilder.prototype, {
    * the vertex shader.
    *
    * @memberof ShaderBuilder.prototype
-   * @type {Object.<String, Number>}
```
Comment (jjhembd) [inline packages/engine/Source/Renderer/ShaderBuilder.js:95]: We still have "Number" here. It is an unusual pattern, perhaps your search pattern missed it?

File: packages/engine/Source/Scene/Cesium3DTilePointFeature.js [hunk 2]
```diff
@@ -732,8 +732,8 @@ Cesium3DTilePointFeature.prototype.hasProperty = function (name) {
  *
  * @see {@link https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_batch_table_hierarchy}
  *
- * @param {String[]} [results] An array into which to store the results.
- * @returns {String[]} The IDs of the feature's properties.
+ * @param {string[]} [results] An array into which to store the results.
+ * @returns {} The IDs of the feature's properties.
```
Comment (jjhembd) [inline packages/engine/Source/Scene/Cesium3DTilePointFeature.js:736]: The return type is missing here?

File: packages/engine/Source/Scene/Cesium3DTileStyle.js [hunk 3]
```diff
@@ -1445,7 +1445,7 @@ Cesium3DTileStyle.prototype.getPointSizeShaderFunction = function (
 /**
  * Gets the variables used by the style.
  *
- * @returns {String[]} The variables used by the style.
+ * @returns {} The variables used by the style.
```
Comment (jjhembd) [inline packages/engine/Source/Scene/Cesium3DTileStyle.js:1448]: Also missing the return type here. Perhaps we need to check for other `{String[]}` types to see if we have a recurring problem.

File: packages/engine/Source/Scene/ConditionsExpression.js [hunk 4]
```diff
@@ -188,7 +188,7 @@ ConditionsExpression.prototype.getShaderFunction = function (
 /**
  * Gets the variables used by the expression.
  *
- * @returns {String[]} The variables used by the expression.
+ * @returns {} The variables used by the expression.
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ConditionsExpression.js:191]: Another place missing the {string[]} return type

File: packages/engine/Source/Scene/DracoLoader.js [hunk 5]
```diff
@@ -54,11 +54,11 @@ DracoLoader.decodePointCloud = function (parameters) {
...
  * Decodes a buffer view. Returns undefined if the task cannot be scheduled.
  *
- * @param {Object} options Object with the following properties:
+ * @param {object} options Object with the following properties:
  * @param {Uint8Array} options.array The typed array containing the buffer view data.
- * @param {Object} options.bufferView The glTF buffer view object.
- * @param {Object.<String, Number>} options.compressedAttributes The compressed attributes.
- * @param {Boolean} options.dequantizeInShader Whether POSITION and NORMAL attributes should be dequantized on the GPU.
+ * @param {object} options.bufferView The glTF buffer view object.
+ * @param {Object.<string, Number>} options.compressedAttributes The compressed attributes.
```
Comment (jjhembd) [inline packages/engine/Source/Scene/DracoLoader.js:60]: `Number` should be `number`.
For Objects where the type of the key and value are specified, the value type has not been updated.

File: packages/engine/Source/Scene/Expression.js [hunk 6]
```diff
@@ -223,7 +223,7 @@ Expression.prototype.getShaderExpression = function (
 /**
  * Gets the variables used by the expression.
  *
- * @returns {String[]} The variables used by the expression.
+ * @returns {} The variables used by the expression.
```
Comment (jjhembd) [inline packages/engine/Source/Scene/Expression.js:226]: Another missed `string[]` return type

File: packages/engine/Source/Scene/ScreenSpaceCameraController.js [hunk 7]
```diff
@@ -591,9 +591,9 @@ function handleZoom(
     return;
   }
 
-  const sameStartPosition = Cartesian2.equals(
-    startPosition,
-    object._zoomMouseStart
+  const sameStartPosition = defaultValue(
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ScreenSpaceCameraController.js:594]: This looks like an unrelated change. Is it needed in this PR?

File: packages/engine/Source/Scene/ModelComponents.js [hunk 8]
```diff
@@ -589,7 +589,7 @@ function Primitive() {
    * The feature IDs associated with this primitive. Feature ID types may
    * be interleaved
    *
-   * @type {Array.<ModelComponents.FeatureIdAttribute|ModelComponents.FeatureIdImplicitRange|ModelComponents.FeatureIdTexture>}
+   * @type {ModelComponents.FeatureIdAttribute[]|ModelComponents.FeatureIdImplicitRange[]|ModelComponents.FeatureIdTexture[]}
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ModelComponents.js:592]: This new syntax doesn't allow Feature ID types to be interleaved. It allows arrays like this:
`[{FeatureIdAttribute}, {FeatureIdAttribute}]`
OR this:
`[{FeatureIdImplicitRange}, {FeatureIdImplicitRange}]`
but the old code also allowed this:
`[{FeatureIdAttribute}, {FeatureIdImplicitRange}]`.
Can you update the syntax here to allow interleaved types? 

I think there may be a couple other places like this one, where we want to allow arrays with elements of different types.

**Review by jjhembd — CHANGES_REQUESTED** [review]
Hi @onsummer, thanks a lot for this PR! It will be good to make our type syntax more consistent. 

I noted a couple smaller issues and questions in the comments above. For each of these, I only flagged a few examples--there may be several other places with the same issue.

File: packages/engine/Source/Scene/ScreenSpaceCameraController.js (hunk 7 above)
Comment (onsummer) [reply packages/engine/Source/Scene/ScreenSpaceCameraController.js:594]: I forgot to remove that, sorry

File: packages/engine/Source/Scene/ModelComponents.js (hunk 8 above)
Comment (onsummer) [reply packages/engine/Source/Scene/ModelComponents.js:592]: Well, this is my careless. I will roll back to `Array<ModelComponents.FeatureIdAttribute|ModelComponents.FeatureIdImplicitRange|ModelComponents.FeatureIdTexture>`.

**Comment by onsummer** [comment]
@jjhembd Thank you for your review. I fix some issues that you mentioned above as possible as I can found, PTAL

File: packages/engine/Source/Scene/ModelComponents.js [hunk 9]
```diff
@@ -643,7 +643,7 @@ function Instances() {
    * The feature ID attributes associated with this set of instances.
    * Feature ID attribute types may be interleaved.
    *
-   * @type {Array.<ModelComponents.FeatureIdAttribute|ModelComponents.FeatureIdImplicitRange>}
+   * @type {ModelComponents.FeatureIdAttribute[]|ModelComponents.FeatureIdImplicitRange[]}
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ModelComponents.js:646]: This one also needs to allow interleaved types

File: packages/engine/Source/Scene/ModelComponents.js [hunk 10]
```diff
@@ -589,7 +589,7 @@ function Primitive() {
    * The feature IDs associated with this primitive. Feature ID types may
    * be interleaved
    *
-   * @type {Array.<ModelComponents.FeatureIdAttribute|ModelComponents.FeatureIdImplicitRange|ModelComponents.FeatureIdTexture>}
+   * @type {Array<ModelComponents.FeatureIdAttribute|ModelComponents.FeatureIdImplicitRangeModelComponents.FeatureIdTexture>}
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ModelComponents.js:592]: I think this is missing a "|"

File: packages/engine/Source/Scene/ScreenSpaceCameraController.js [hunk 11]
```diff
@@ -2187,7 +2189,7 @@ function zoom3D(controller, startPosition, movement) {
     camera.position,
     zoom3DCartographic
   ).height;
-  if (height < controller._minimumPickingTerrainHeight) {
+  if (height < controller._minimumPickingTerrainHeight && !inertiaMovement) {
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ScreenSpaceCameraController.js:2192]: This looks like another unrelated change

File: packages/engine/Source/Scene/ScreenSpaceCameraController.js [hunk 12]
```diff
@@ -593,7 +593,8 @@ function handleZoom(
 
   const sameStartPosition = Cartesian2.equals(
     startPosition,
-    object._zoomMouseStart
+    object._zoomMouseStart,
+    Cartesian2.equals(startPosition, object._zoomMouseStart)
```
Comment (jjhembd) [inline packages/engine/Source/Scene/ScreenSpaceCameraController.js:597]: The equality check is duplicated here

File: Documentation/Contributors/DocumentationGuide/README.md [hunk 13]
```diff
@@ -287,53 +287,53 @@ function Cartesian3(x, y) {
...
     this.x = defaultValue(x, 0.0);
 
     // ...
 ```
 
-- Use `@memberOf` when documenting property getter/setters, e.g.,
+- Use `@memberof` when documenting property getter/setters, e.g.,
 
 ```javascript
 Object.defineProperties(Entity.prototype, {
```
Comment (jjhembd) [inline Documentation/Contributors/DocumentationGuide/README.md:301]: Thanks for catching all these things in a README!

**Review by jjhembd — APPROVED** [review]

**Comment by jjhembd** [comment]
Hi @onsummer, thanks for those changes! 
I pointed out a few small remaining things in `ModelComponents` and `ScreenSpaceCameraController`. Once those are done, this will be ready to merge.

**Comment by jjhembd** [comment]
Hi @onsummer, thanks again for this contribution. I learned something about how to write better JSDoc just by looking at your edits.

We would like to make sure future code changes follow the same pattern. We have an open issue to automate JSDoc checks in #10815. Once this PR is merged, would you be interested in contributing to that issue?

**Comment by onsummer** [comment]
> Hi @onsummer, thanks again for this contribution. I learned something about how to write better JSDoc just by looking at your edits.
> 
> We would like to make sure future code changes follow the same pattern. We have an open issue to automate JSDoc checks in #10815. Once this PR is merged, would you be interested in contributing to that issue?

That seems a pretty challenge to me, and I am interesting in contributing it.

**Comment by onsummer** [comment]
> Hi @onsummer, thanks for those changes! I pointed out a few small remaining things in `ModelComponents` and `ScreenSpaceCameraController`. Once those are done, this will be ready to merge.

:) Maybe this time will be OK to merge

**Comment by jjhembd** [comment]
Looks good, thanks a lot @onsummer! Merging now
